tests/test_elegant_conversion.py

```diff
@@ -32,14 +33,42 @@ def test_fodo():
     assert [element.name for element in converted.elements] == [
         element.name for element in correct_lattice.elements
     ]
-    assert converted.q1.length == correct_lattice.q1.length
-    assert converted.q1.k1 == correct_lattice.q1.k1
-    assert converted.q2.length == correct_lattice.q2.length
-    assert converted.q2.k1 == correct_lattice.q2.k1
-    assert [d.length for d in converted.d1] == [d.length for d in correct_lattice.d1]
-    assert converted.d2.length == correct_lattice.d2.length
-    assert converted.s1.length == correct_lattice.s1.length
-    assert converted.s1.e1 == correct_lattice.s1.e1
+    assert torch.isclose(converted.q1.length, correct_lattice.q1.length)
+    assert torch.isclose(converted.q1.k1, correct_lattice.q1.k1)
+    assert torch.isclose(converted.q2.length, correct_lattice.q2.length)
+    assert torch.isclose(converted.q2.k1, correct_lattice.q2.k1)
+    for i in range(2):
+        assert torch.isclose(converted.d1[i].length, correct_lattice.d1[i].length)
+    assert torch.isclose(converted.d2.length, correct_lattice.d2.length)
+    assert torch.isclose(converted.s1.length, correct_lattice.s1.length)
+    assert torch.isclose(converted.s1.e1, correct_lattice.s1.e1)
+
+
+def test_cavity_import():
+    """Test importing an accelerating cavity defined in the Elegant file format."""
```

Comment:
I feel like this is a weird instance of cavity import and the docstring (and maybe even the name) of the test should reflect and explain that.

Reply:
On second thoughts, this is really two separate tests, one testing if the cavity was imported correctly (and without throwing an exception) and another doing the save for a custom transfer map, right?

Reply:
I`ve decided to split the test into two separate parts. One for testing the regular cavity import, the second for the custom transfer map.